### PR TITLE
Silencerole

### PR DESCRIPTION
**Pull request by FordFriedel**
Allows server admins to designate a role as 'silenced'

Adds a silenceCheck for individual slash commands. If true it will check a user if they are silenced or if they have the silenced role.

Visual update to /config

### CODE REVIEW

File: src/slashcommands/SilenceRole.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            name: 'role',
+            description: 'The role to silence.',
+            type: ApplicationCommandOptionTypes.ROLE,
+            required: true,
+        }]
+    }
+    requiredPermissions: bigint[] = [];
+    run(bot: Bot, interaction: CommandInteraction<CacheType>): Promise<void> {
+        try{
+            //TODO if someone attempts to set the same role that is already set, remove it
```
Comment (CPhantasm) [inline src/slashcommands/SilenceRole.ts:26]: Let's try to tackle TODOs before submitting PRs. There's a draft feature for a reason.

File: src/slashcommands/SilenceRole.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+				content: 'Error: contact a developer to investigate',
+				ephemeral: true,
+			});
+		}
+    }
+    guildRequired?: boolean | undefined = true;
+    managerRequired?: boolean | undefined = true;
+}
+
+export async function silenceCheck(interaction: Interaction): Promise<boolean> {
```
Comment (CPhantasm) [inline src/slashcommands/SilenceRole.ts:67]: Does this have to be an async function? I don't see you performing and asynchronous operations in it

File: src/slashcommands/SilenceRole.ts (hunk 1 above)
Comment (FordFriedel) [reply src/slashcommands/SilenceRole.ts:26]: resolved this already, forgot to remove the /TODO

File: src/slashcommands/SilenceRole.ts (hunk 2 above)
Comment (FordFriedel) [reply src/slashcommands/SilenceRole.ts:89]: image

When I remove the async I get this

**Pull request merged**